.github/workflows/mariner2.yml

```diff
@@ -0,0 +1,45 @@
+name: build for mariner2
+
+on:
+  push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l -o pipefail {0}
+    container:
+      image: mcr.microsoft.com/cbl-mariner/base/core:2.0
+    steps:
+      - run: |
+          tdnf install --noplugins --skipsignature -y ca-certificates git
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          set-safe-directory: true
+      - name: install dependencies
+        run: |
+          bash .install/mariner2.sh
+      - name: get redis latest
+        run: |
+          bash .install/get_redis.sh
+      - name: build
+        run: |
+          make build
+      - name: test
+        run: |
+          make test
+      - name: pack module
+        run: |
+          make pack BRANCH=${{ github.ref_name }}
```

Comment:
why did you add `BRANCH=${{ github.ref_name }}`?

Reply:
I was getting this error of permissions when pack.sh tries to get the branch name, and altough I tried to solve it in the checkout action it did not work.

```
fatal: detected dubious ownership in repository at '/__w/RedisJSON/RedisJSON'
To add an exception for this directory, call:

	git config --global --add safe.directory /__w/RedisJSON/RedisJSON

In /__w/RedisJSON/RedisJSON/sbin/pack.sh:
239      #----------------------------------------------------------------------------------------------
240      
241      if [[ -z $BRANCH ]]; then
242  >>> 	BRANCH=$(git rev-parse --abbrev-ref HEAD)
243      	# this happens of detached HEAD
244      	if [[ $BRANCH == HEAD ]]; then
245      		BRANCH="$SEMVER"

make: *** [Makefile:231: pack] Error 1
Error: Process completed with exit code 2.
```